src/dlgConnectionProfiles.cpp

```diff
@@ -2299,40 +2299,30 @@ void dlgConnectionProfiles::loadProfile(bool alsoConnect)
-            || pHost->getUrl().contains(QStringLiteral("achaea.com"), Qt::CaseInsensitive)
-            || pHost->getUrl().contains(QStringLiteral("lusternia.com"), Qt::CaseInsensitive)
-            || pHost->getUrl().contains(QStringLiteral("imperian.com"), Qt::CaseInsensitive)) {
-            // IRE MUD profiles:
-            mudlet::self()->packagesToInstallList.append(QStringLiteral(":/mudlet-mapper.xml"));
-            mudlet::self()->packagesToInstallList.append(QStringLiteral(":/send-text-to-all-games.xml"));
-            mudlet::self()->packagesToInstallList.append(QStringLiteral(":/deleteOldProfiles.xml"));
-            mudlet::self()->packagesToInstallList.append(QStringLiteral(":/echo.xml"));
-
-        } else if (pHost->getUrl().contains(QStringLiteral("carrionfields.net"), Qt::CaseInsensitive)) {
-            // Carrion Fields loader - downloads and installs latest GUI from GitHub site
-            mudlet::self()->packagesToInstallList.append(QStringLiteral(":/CF-loader.xml"));
-            mudlet::self()->packagesToInstallList.append(QStringLiteral(":/send-text-to-all-games.xml"));
-            mudlet::self()->packagesToInstallList.append(QStringLiteral(":/deleteOldProfiles.xml"));
-            mudlet::self()->packagesToInstallList.append(QStringLiteral(":/echo.xml"));
-
-        } else if (pHost->getUrl().contains(QStringLiteral("mudlet.org"), Qt::CaseInsensitive)) {
-            // Mudlet self-test profile ONLY:
-            mudlet::self()->packagesToInstallList.append(QStringLiteral(":/run-tests.xml"));
+    if (preIntallPackages) {
+        auto gameUrl = pHost->getUrl().toLower();
+        const QHash<QString, QStringList> defaultScripts = {
+                {QStringLiteral(":/run-lua-code-v4.xml"), {QStringLiteral("*")}},
+                {QStringLiteral(":/echo.xml"), {QStringLiteral("*")}},
+                {QStringLiteral(":/send-text-to-all-games.xml"), {QStringLiteral("*")}},
+                {QStringLiteral(":/deleteOldProfiles.xml"), {QStringLiteral("*")}},
+                {QStringLiteral(":/CF-loader.xml"), {QStringLiteral("carrionfields.net")}},
+                {QStringLiteral(":/run-tests.xml"), {QStringLiteral("mudlet.org")}},
+                {QStringLiteral(":/mudlet-mapper.xml"),
+                 {QStringLiteral("aetolia.com"), QStringLiteral("achaea.com"), QStringLiteral("lusternia.com"), QStringLiteral("imperian.com"), QStringLiteral("starmourn.com")}},
+        };
+
+        QHashIterator<QString, QStringList> i(defaultScripts);
+        while (i.hasNext()) {
+            i.next();
+            if (i.value().first() == QLatin1String("*") || i.value().contains(gameUrl)) {
+                mudlet::self()->packagesToInstallList.append(i.key());
+            }
+        }
 
-        } else {
-            // All OTHER profiles:
+        if (!mudlet::self()->packagesToInstallList.contains(QStringLiteral(":/mudlet-mapper.xml"))) {
```

Comment:
:warning: Incorrect logic - the "mudlet.org" profile did not load either mapper before.

Reply:
The change is intended.